src/lib.rs

```diff
@@ -155,26 +156,33 @@ impl TextEmbedding {
 
         let threads = available_parallelism()?.get() as i16;
 
-        let model_repo =
-            TextEmbedding::retrieve_model(model_name.clone(), cache_dir, show_download_progress)?;
+        let model_repo = TextEmbedding::retrieve_model(
+            model_name.clone(),
+            cache_dir.clone(),
+            show_download_progress,
+        )?;
 
-        // The model files could be placed in subdirectories
-        let model_file = model_repo
-            .info()?
-            .siblings
-            .into_iter()
-            .find(|f| {
-                f.rfilename.ends_with("model.onnx") || f.rfilename.ends_with("model_optimized.onnx")
-            })
-            .unwrap();
+        let model_file_info_result = model_repo.info();
 
-        let model_file = model_repo.get(&model_file.rfilename)?;
+        // If the attempt fails (likely no connection), fall back on cached onnx file
+        let model_file_reference = match model_file_info_result {
+            std::result::Result::Ok(info) => {
+                TextEmbedding::retrieve_remote_model_file(info, &model_repo)
+            }
+            Err(ref e) => {
+                eprintln!("Error retrieving model info (likely no connection): {}", e);
```

Comment:
Probably remove this line.
The error message looks something like
```console
Error retrieving model info (likely no connection): request error: https://huggingface.co/api/models/Qdrant/bge-small-en-v1.5-onnx-Q/revision/main: Dns Failed: resolve dns name 'huggingface.co:443': failed to lookup address information: nodename nor servname provided, or not known
```

Which appears like something terrible happened. When instead it was properly handled.

Reply:
Will do! Thanks